### PR TITLE
ApiTest — SQL -> AR/Arel

### PR DESCRIPTION
**Pull request by nimmolo**
Converts all test method `where` clauses to AR with arel predications

### CODE REVIEW

File: test/models/api_test.rb [hunk 1]
```diff
@@ -1727,9 +1755,9 @@ def test_getting_names
     assert_api_pass(params.merge(rank: "variety"))
     assert_api_results(names)
 
-    with    = Name.where.not("author is null or author = ''").
+    with    = Name.where.not(Name[:author].blank).
               reject(&:correct_spelling_id)
-    without = Name.where("author is null or author = ''").
+    without = Name.where(Name[:author].blank).
```
Comment (pellaea) [inline test/models/api_test.rb:1760]: Neat!  Is Model[:col].blank identical to the SQL "col is null or col = ''"??  Super useful!

File: test/models/api_test.rb [hunk 2]
```diff
@@ -2206,37 +2242,43 @@ def test_getting_observations
...
+    obses = Observation.where(
+      (Observation[:lat] >= 34).and(Observation[:lat] <= 35).
+      and(Observation[:long] >= -119).and(Observation[:long] <= -118)
+    )
+    locs = Location.where(
+      (Location[:south] >= 34).and(Location[:north] <= 35).
+      and(Location[:west] >= -119).and(Location[:east] <= -118).
+      and(Location[:west] <= Location[:east])
+    )
     obses = (obses + locs.map(&:observations)).flatten.uniq.sort_by(&:id)
```
[the remark this answers is not in the document]
Comment (nimmolo) [reply test/models/api_test.rb:2282]: Ohh, yes!

File: test/models/api_test.rb (hunk 1 above)
Comment (nimmolo) [reply test/models/api_test.rb:1760]: Yes. Those arel-extensions are so well written. `blank` does a `trim(coalesce(''))`

File: test/models/api_test.rb [hunk 3]
```diff
@@ -2842,7 +2892,9 @@ def test_getting_sequences
                        archive: "GenBank", accession: "MO#{obs.id}")
     end
 
-    obses = Observation.where("year(`when`) >= 2012 and year(`when`) <= 2014")
+    obses = Observation.where(
+      (Observation[:when].year >= 2012).and(Observation[:when].year <= 2014)
+    )
```
[the remark this answers is not in the document]
Comment (nimmolo) [reply test/models/api_test.rb:2897]: Gonna try it!

[the remark this answers is not in the document]
Comment (nimmolo) [reply test/models/api_test.rb:2892]: Needs the `year`. Syntax is `Observation.where(Observation[:when].year.between(2012..2014))`

**Pull request merged**